### PR TITLE
runtime: gate nonce disable at restart slot

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

Need to play durable nonce tx normally through the restart slot

#### Summary of Changes

gate durable nonce disable at slot 135986379

### CODE REVIEW

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -4126,10 +4126,13 @@ impl Bank {
 
     pub fn check_transaction_for_nonce(
         &self,
-        _tx: &SanitizedTransaction,
+        tx: &SanitizedTransaction,
     ) -> Option<TransactionAccount> {
-        None
-        // self.check_message_for_nonce(tx.message())
+        if self.slot() > 135986379 {
```
Comment (mvines) [inline runtime/src/bank.rs:4131]: Can we wrap the entire logic with a `self.cluster_type() == ClusterType::MainnetBeta`

Comment (CriesofCarrots) [inline runtime/src/bank.rs:4131]: Is this right? I thought we wanted to preserve behavior before that slot

Comment (CriesofCarrots) [reply runtime/src/bank.rs]: Yes plz

Comment (t-nelson) [reply runtime/src/bank.rs]: done

**Review by mvines — APPROVED** [review]

**Review by CriesofCarrots — APPROVED** [review]
There's a simpler way to write it, but meh

**Pull request merged**